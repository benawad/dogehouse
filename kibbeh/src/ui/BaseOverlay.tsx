import React, { MouseEventHandler, ReactNode } from "react";
import ProfileHeaderWrapperStory from "../stories/ProfileHeaderWrapper.story";

export interface BaseOverlayProps
  extends React.ComponentPropsWithoutRef<"div"> {
  title?: string;
  actionButton?: string;
  onActionButtonClicked?: MouseEventHandler<HTMLButtonElement>;
  children: ReactNode;
  overlay?: ReactNode;
}

export const BaseOverlay: React.FC<BaseOverlayProps> = ({
  children,
  title,
  actionButton,
  overlay,
  onActionButtonClicked,
  ...props
}) => {
  return (
    <div
      className="flex flex-col w-full rounded-8 bg-primary-800 border border-primary-700 overflow-hidden relative"
      data-testid="base-overlay"
      {...props}
    >
      {overlay ? overlay : ""}
      {title && (
        <div className="px-4 py-3 border-b border-primary-600 flex items-center">
          <h4 className="text-primary-100">{title}</h4>
        </div>
      )}

      <div className="flex flex-col text-primary-100">{children}</div>

      {actionButton && (
        <button
          className="flex px-4 bg-primary-700 text-primary-100 outline-none font-bold"
          style={{
            paddingTop: 8,
            paddingBottom: 12,
            borderRadius: "0 0 8px 8px",
          }}
          onClick={onActionButtonClicked}
        >
          {actionButton}
        </button>
      )}
    </div>
  );
};
