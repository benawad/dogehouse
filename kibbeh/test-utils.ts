import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
