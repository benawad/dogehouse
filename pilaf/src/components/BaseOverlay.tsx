import React, { ReactNode } from "react";
import {
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  ViewStyle,
} from "react-native";
import { colors, fontFamily, fontSize, radius } from "../constants/dogeStyle";

interface BaseOverlayProps {
  style?: ViewStyle;
  title?: string;
  actionButton?: string;
  onActionButtonClicked?: () => void;
  children: ReactNode;
}

export const BaseOverlay: React.FC<BaseOverlayProps> = ({
  children,
  title,
  style,
  actionButton,
  onActionButtonClicked,
}) => {
  return (
    <View style={[style, styles.container]}>
      {title && (
        <View style={styles.header}>
          <Text style={styles.title}>{title}</Text>
        </View>
      )}
      <ScrollView>{children}</ScrollView>
      {actionButton && (
        <TouchableOpacity style={styles.button} onPress={onActionButtonClicked}>
          <Text style={styles.buttonTitle}>{actionButton}</Text>
        </TouchableOpacity>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.primary800,
    borderRadius: radius.m,
  },
  header: {
    borderBottomWidth: 1,
    borderColor: colors.primary600,
    padding: 16,
  },
  title: {
    color: colors.text,
    fontFamily: fontFamily.regular,
    fontSize: fontSize.h4,
    fontWeight: "700",
  },
  button: {
    backgroundColor: colors.primary700,
    padding: 16,
    borderBottomLeftRadius: radius.m,
    borderBottomRightRadius: radius.m,
  },
  buttonTitle: {
    fontSize: fontSize.paragraph,
    fontFamily: fontFamily.bold,
    color: colors.text,
  },
});
