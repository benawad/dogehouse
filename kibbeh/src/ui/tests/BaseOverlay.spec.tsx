import React from "react";

import { render } from "../../../test-utils";
import { BaseOverlay } from "../BaseOverlay";

describe("BaseOverlay", () => {
  it("should render correctly", () => {
    const { getByTestId, getByText } = render(
      <BaseOverlay>
        <div className="flex">Child</div>
      </BaseOverlay>
    );
    const baseOverlay = getByTestId("base-overlay");
    const child = getByText("Child");

    expect(baseOverlay).toBeVisible();
    expect(child).toBeVisible();
  });

  it("should render title if title prop is passed", () => {
    const { getByText } = render(
      <BaseOverlay title="Messages">
        <div className="flex">Child</div>
      </BaseOverlay>
    );
    const title = getByText("Messages");

    expect(title).toBeVisible();
  });

  it("should render actionButton if actionButton prop is passed", () => {
    const { getByText } = render(
      <BaseOverlay actionButton="Show more">
        <div className="flex">Child</div>
      </BaseOverlay>
    );
    const actionButton = getByText("Show more");

    expect(actionButton).toBeVisible();
  });

  it("should render call onActionButtonClicked if actionButton is clicked", () => {
    const onActionButtonClickedMock = jest.fn();
    const { getByText } = render(
      <BaseOverlay
        actionButton="Show more"
        onActionButtonClicked={onActionButtonClickedMock}
      >
        <div className="flex">Child</div>
      </BaseOverlay>
    );
    const actionButton = getByText("Show more");

    actionButton.click();

    expect(onActionButtonClickedMock).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    const { getByTestId } = render(
      <BaseOverlay>
        <div className="flex">Child</div>
      </BaseOverlay>
    );
    const baseOverlay = getByTestId("base-overlay");

    expect(baseOverlay).toMatchSnapshot();
  });
});
