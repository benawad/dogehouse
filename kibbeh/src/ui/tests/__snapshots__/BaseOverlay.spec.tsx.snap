// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseOverlay should match snapshot 1`] = `
<div
  class="flex flex-col w-full rounded-8 bg-primary-800 border border-primary-700 overflow-hidden relative"
  data-testid="base-overlay"
>
  
  <div
    class="flex flex-col text-primary-100"
  >
    <div
      class="flex"
    >
      Child
    </div>
  </div>
</div>
`;
